src/librustdoc/doctest.rs

```diff
@@ -713,6 +713,22 @@ impl IndividualTestOptions {
     }
 }
 
+fn beautify_path(path: &Path) -> PathBuf {
+    let mut beautiful_path = PathBuf::new();
+
+    for component in path.components().filter(|c| !matches!(c, Component::CurDir)) {
+        if matches!(component, Component::ParentDir) {
+            if beautiful_path.parent().is_none() {
+                return path.canonicalize().unwrap_or_else(|_| path.to_path_buf());
```

Comment:
Calling canonicalize at the end seems to defeat the point of manually adjusting the path, and is what `@fasterthanlime` advocated against. Did yoi misread their comment?